Helm chart lint: bump version on release or change GHA

Whenever we do a release, the helm chart linting GHA fails because it expects us to increase the version of the helm chart.

Solution options:
- Update version of helm chart automatically or manually when merging a release back into dev, or, 
- Only run linting on PRs that actually change the helm chart similar to flake8.

For easier access, error merging back to `dev` was:

```
Run ct lint --config ct.yaml
Linting charts...

------------------------------------------------------------------------------------------------------------------------
 Charts to be processed:
------------------------------------------------------------------------------------------------------------------------
 defectdojo => (version: "1.5.1", path: "helm/defectdojo")
------------------------------------------------------------------------------------------------------------------------

Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "bitnami" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 4 charts
Downloading mysql from repo https://charts.helm.sh/stable
Downloading postgresql from repo https://charts.bitnami.com/bitnami
Downloading rabbitmq from repo https://charts.bitnami.com/bitnami
Downloading redis from repo https://charts.bitnami.com/bitnami
Deleting outdated charts
Linting chart 'defectdojo => (version: "1.5.1", path: "helm/defectdojo")'
Checking chart 'defectdojo => (version: "1.5.1", path: "helm/defectdojo")' for a version bump...
Error: Error linting charts: Error processing charts
Old chart version: 1.5.1
New chart version: 1.5.1
------------------------------------------------------------------------------------------------------------------------
 ✖︎ defectdojo => (version: "1.5.1", path: "helm/defectdojo") > Chart version not ok. Needs a version bump!
------------------------------------------------------------------------------------------------------------------------
Error linting charts: Error processing charts
Error: Process completed with exit code 1.
```

--- a/.github/workflows/test-helm-chart.yml
+++ b/.github/workflows/test-helm-chart.yml
@@ -40,13 +40,20 @@ jobs:
       - name: Run chart-testing (list-changed)
         id: list-changed
         run: |
-          changed=$(ct list-changed --config ct.yaml)
+          if [ ! -z ${GITHUB_BASE_REF} ]; then
+            echo "pull_request: ct list-changed --config ct.yaml --target-branch ${GITHUB_BASE_REF}"
+            changed=$(ct list-changed --config ct.yaml --target-branch ${GITHUB_BASE_REF})
+          else
+            echo "push: ct list-changed --config ct.yaml --target-branch ${GITHUB_REF#refs/heads/}"
+            changed=$(ct list-changed --config ct.yaml --target-branch ${GITHUB_REF#refs/heads/})
+          fi
           if [[ -n "$changed" ]]; then
             echo "::set-output name=changed::true"
           fi
 
       - name: Run chart-testing (lint)
         run: ct lint --config ct.yaml
+        if: steps.list-changed.outputs.changed == 'true'
 
       - name: Create kind cluster
         uses: helm/kind-action@v1.1.0
@@ -54,3 +61,4 @@ jobs:
 
       - name: Run chart-testing (install)
         run: ct install
+        if: steps.list-changed.outputs.changed == 'true'

